File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.js
/**
 * @param {number} n
 * @param {number} t
 * @return {number}
 */
var smallestNumber = function(n, t) {
    const check=(x)=>{
        let ans=1;
        while(x){
            ans*=x%10;
            x=Math.floor(x/10);
        }
        return ans;
    }
    for(let i=n;i<=100;i++){
        const x=check(i);
        if(x%t==0) {
            return i;
        }
    }
};

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        for(int i=n;i<=100;i++){
            int x=check(i);
            if(x%t==0) {
                return i;
            }
        }
        return 0;
    }
    int check(int x) {
        int ans=1;
        while(x){
            ans*=x%10;
            x/=10;
        }
        return ans;
    }
};